src/server/page_server.cpp

```diff
@@ -388,7 +388,9 @@ page_server::start_log_replicator (const log_lsa &start_lsa)
   assert (is_page_server ());
   assert (m_replicator == nullptr);
 
-  m_replicator.reset (new cublog::replicator (start_lsa));
+  const int replication_parallel_count = prm_get_integer_value (PRM_ID_REPLICATION_PARALLEL_COUNT);
+  assert (replication_parallel_count >= 0);
+  m_replicator.reset (new cublog::replicator (start_lsa, RECOVERY_PAGE, replication_parallel_count));
```

Comment:
parameterized replicator to be able to instrument it differently for PS and PTS

Reply:
Another way to look at it is to keep characteristics independent of whether replication is running in page server or passive transaction server context. Because some of these parameters (eg: page fetch type) will be forwarded down to the recovery functions on which replication is built on.
Of course that we could fork these params inside the ctor, or pass them from the outside.